crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -633,7 +677,7 @@ fn compare_error_fn(
             let fidelity_product: f64 = circuit
                 .gates
                 .iter()
-                .map(|x| 1. - err_map.get(&x.0).unwrap_or(&0.))
+                .map(|x| 1. - err_map.get(x.0.name()).unwrap_or(&0.))
```

Comment:
Very minor, but would save looking up for context. I with clippy; no change in linewrap!
```suggestion
                .map(|gate| 1. - err_map.get(gate.0.name()).unwrap_or(&0.))
```

Reply:
Have we considered using a non-string key? (Or some kind of [optimized string](https://swatinem.de/blog/optimized-strings/))

Reply:
Not really, these strings are lookup keys from the `Target`, the error map here is just a rust view of the error data in the target that we construct in Python to provide the data to this pass (when we have #12292 merged we can drop this structure and query it directly). We need something that interoperates with a Python string which is realistically just `String` or `&str`.

Reply:
I assumed that, at least for now, we need regular strings for Python interop.